ZenPacks/zenoss/ZenPackLib/lib/spec/Spec.py

```diff
@@ -171,6 +172,55 @@ def __str__(self):
 
         return "{}({})".format(self.__class__.__name__, ' - '.join(parts))
 
+    @property
+    def order(self):
+        """order attribute used by ClassSpec, ClassRelationSpec, ClassPropertySpec"""
+        return self._order
+
+    @order.setter
+    def order(self, value):
+        """order should be an integer between 1 and 100"""
+        # deal with legacy cases
+        if isinstance(value, int):
+            self._order = int(max(0, min(100, value)))
+        else:
+            self._order = value
+
+    @property
+    def containing_spec(self):
+        """Return Spec containing this Spec"""
+        return None
+
+    @property
+    def normalize_order(self):
+        """Return True if order should be renormalized"""
+        if isinstance(self.order, float):
+            return True
+        return False
+
+    def normalize_children(self, items):
+        """Return True if order should be normalized"""
+        for i in items:
+            if i.normalize_order:
+                return True
+        return False
+
+    def normalize_child_order(self, items):
+        """Deal with legacy order parameters given as float"""
+        if self.normalize_children(items):
+            candidates = [p for p in items if p.order != 50]
+            values = [c.order for c in candidates]
+            factor = .5
```

Comment:
This normalization doesn't work very well for at least the CiscoUCS ZenPack. Many different floats that are intentionally larger and smaller than each other get normalized to the same integer values.

For example:
- 4.1 -> 45
- 4.2 -> 45
- 4.3 -> 45
- 4.4 -> 45
- 4.5 -> 57
- 4.5 -> 57

There's a typical approach to changing the scale of a set of numbers that would fare a little better, but it's still subject to duplicate results for very-close inputs.

```python
def rescale(values, new_min=1, new_max=100):
    if len(values) < 2:
        return map(int, values)
    old_min = min(values)
    old_max = max(values)
    a = (new_max - new_min) / (old_max - old_min)
    b = new_max - a * old_max
    return [a * x + b for x in values]
```

A more fool-proof approach for this particular case is probably to just sort then sequence.

```python
new_orders = itertools.chain(xrange(1, 101), itertools.repeat(100))
for c in sorted(candidates, key=operator.attrgetter("order"):
    c.order = new_orders.next()
```